view/rooms/app.js

```diff
@@ -127,48 +139,178 @@ function createFooter(bot) {
+ *   javascript dom as key value pairs
  */
 function parseBot(bot) {
-  // Unzip bots
-  const zip = new AdmZip(new Buffer(bot.ui.data));
-  const zipEntries = zip.getEntries();
   const botScript = document.createElement('script');
   botScript.id = bot.name + '-script';
 
+  const contentSection = document.createElement('div');
+  contentSection.className = 'slds-section__content';
+
+  try {
+    // Unzip bots
+    const zip = new AdmZip(new Buffer(bot.ui.data));
+    const zipEntries = zip.getEntries();
+
+    // 'index.html' contains root elements that scripts hook up to
+    // and needs to be loaded into the DOM first
+    const index = zipEntries.filter((entry) => entry.name === 'index.html');
+    if (index.length > ZERO) {
+      contentSection.innerHTML = zip.readAsText(index[ZERO]);
+      const elem = contentSection
+        .querySelector('script[src="/index_bundle.js"]');
+      if (elem) {
+        elem.parentNode.removeChild(elem);
+      }
+    }
+
+    // go through zipEntries that arent 'index.html'
+    zipEntries.filter(
+      (entry) => entry.name !== 'index.html'
+    ).forEach((script) => {
+      botScript.appendChild(
+        document.createTextNode(zip.readAsText(script))
+      );
+    });
+
+    const output = {
+      'html': contentSection,
+      'js': botScript
+    };
+    return output;
+  } catch (exception) {
+    debugMessage('Parse Bot failed', exception);
+    return {};
+  }
+} // parseBots
+
+/**
+ * Create and fills DOM containers for bots and adds them to
+ * to the page.
+ *
+ * @param {Object} bot - Bot response with UI
+ */
+function displayBot(bot) {
   // Get the bots section of the page
   const botContainer = document.createElement('div');
   botContainer.id = bot.name + '-section';
   botContainer.className = 'slds-large-size--1-of-3';
-  const contentSection = document.createElement('div');
-  contentSection.className = 'slds-section__content';
+
   const headerSection = createHeader(bot);
   const footerSection = createFooter(bot);
 
-  // 'index.html' contains root elements that scripts hook up to
-  // and needs to be loaded into the DOM first
-  const index = zipEntries.filter((entry) => entry.name === 'index.html');
-  if (index.length > ZERO) {
-    contentSection.innerHTML = zip.readAsText(index[ZERO]);
-    headerSection.appendChild(contentSection);
-    headerSection.appendChild(footerSection);
-    botContainer.appendChild(headerSection);
-    botsContainer.appendChild(botContainer);
+  const iframe = document.createElement('iframe');
+  iframe.id = bot.name + '-iframe-section';
+  iframe.style.display = 'block';
+  iframe.style.border = 'none';
+  iframe.style.width = '100%';
+  iframe.frameBorder = 0;
+
+  headerSection.appendChild(iframe);
+  headerSection.appendChild(footerSection);
+  botContainer.appendChild(headerSection);
+  botsContainer.appendChild(botContainer);
+
+  const parsedBot = parseBot(bot);
+  iframeBot(iframe, bot, parsedBot, user);
```

Comment:
@tausifmuzaffar just wondering where user is defined, should this be _user?

Reply:
+1

Reply:
user is defined in the index.pug --- cant use _user because it sterilizes  the data, and the bot needs the unsterilize data

Reply:
@tausifmuzaffar makes sense now, thank you

Reply:
I have found it to be very useful to add a comment saying it's defined in the pug when you use a var defined in the pug, otherwise I find myself searching around for where it's declared, where value is assigned, etc.